### ISSUE TITLE
Can't load on Purpur 1.20.6

### ISSUE THREAD

**Issue by PurpleEastZQ** [Andavin/Images#109]
[23:26:54 ERROR]: [ModernPluginLoadingStrategy] Could not load plugin 'images-2.5.1.jar' in folder 'plugins\.paper-remapped'
org.bukkit.plugin.InvalidPluginException: java.lang.ExceptionInInitializerError
        at io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider.createInstance(SpigotPluginProvider.java:129) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        at io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider.createInstance(SpigotPluginProvider.java:35) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        at io.papermc.paper.plugin.entrypoint.strategy.modern.ModernPluginLoadingStrategy.loadProviders(ModernPluginLoadingStrategy.java:116) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        at io.papermc.paper.plugin.storage.SimpleProviderStorage.enter(SimpleProviderStorage.java:38) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        at io.papermc.paper.plugin.entrypoint.LaunchEntryPointHandler.enter(LaunchEntryPointHandler.java:36) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        at org.bukkit.craftbukkit.CraftServer.loadPlugins(CraftServer.java:546) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:320) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1150) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:326) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.ExceptionInInitializerError
        at java.base/java.lang.Class.forName0(Native Method) ~[?:?]
        at java.base/java.lang.Class.forName(Class.java:534) ~[?:?]
        at java.base/java.lang.Class.forName(Class.java:513) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.<init>(PluginClassLoader.java:78) ~[purpur-api-1.20.6-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider.createInstance(SpigotPluginProvider.java:125) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        ... 9 more
Caused by: java.lang.UnsupportedOperationException: Class class com.andavin.images.PacketListener is not currently supported for version v1_20_R5
        at images-2.5.1.jar/com.andavin.images.Versioned.getInstance(Versioned.java:60) ~[images-2.5.1.jar:?]
        at images-2.5.1.jar/com.andavin.images.Images.<clinit>(Images.java:97) ~[images-2.5.1.jar:?]
        at java.base/java.lang.Class.forName0(Native Method) ~[?:?]
        at java.base/java.lang.Class.forName(Class.java:534) ~[?:?]
        at java.base/java.lang.Class.forName(Class.java:513) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.<init>(PluginClassLoader.java:78) ~[purpur-api-1.20.6-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider.createInstance(SpigotPluginProvider.java:125) ~[purpur-1.20.6.jar:1.20.6-2233-0d6766e]
        ... 9 more

**Comment by CloudeLecaw** [comment]
Update thr plugin. your using 2.5.1..

**Comment by CloudeLecaw** [comment]
no

**Comment by masood151117** [comment]
> use Version 2.4.0 that was made for 1.20.6 for paper or spigot if it still refuse to load then you cant use it on pupur..
> 
> you also can try out version 2.5.2 https://github.com/Kici33/Images?tab=readme-ov-file

no 1.20.6 purpur support . 2.4.0 and 2.5.2 and 2.5.3

**Comment by masood151117** [comment]
@Andavin 
please fix that issue . all my servers are 1.20.6

**Comment by masood151117** [comment]
do you have a discord server ?

**Comment by masood151117** [comment]
@resi23

**Comment by resi23** [comment]
Yes I have a discord but it's not there to help people with their servers and plugin problems.

**Comment by masood151117** [comment]
> Yes I have a discord but it's not there to help people with their servers and plugin problems.

if you can add 2.3.3 version in your github . i cant find it

**Comment by resi23** [comment]
I don't have it anymore because the developer himself fixed the bug for Paperserver in [v2.5.0](https://github.com/Andavin/Images/commit/7d5b87c1ce590a873896d67586dd14fcb8f27444)

```
This has no changes for Spigot since 2.5.0 would work with Spigot 1.21.1.

For Paper, this update adds an extra compatibility layer to revert to the previous minor version if one isn't officially supported. This will save the need for an update when there is a no-op change like 1.21.1.

When there is a substantial change, errors may occur in which case an update will be required on both Spigot and Paper.```

**Comment by masood151117** [comment]
> I don't have it anymore because the developer himself fixed the bug for Paperserver in [v2.5.0](https://github.com/Andavin/Images/commit/7d5b87c1ce590a873896d67586dd14fcb8f27444)
> 
> ```
> This has no changes for Spigot since 2.5.0 would work with Spigot 1.21.1.
> 
> For Paper, this update adds an extra compatibility layer to revert to the previous minor version if one isn't officially supported. This will save the need for an update when there is a no-op change like 1.21.1.
> 
> When there is a substantial change, errors may occur in which case an update will be required on both Spigot and Paper.```
> ```

no version working . visit spigotmc reviews page :/

**Comment by masood151117** [comment]
@resi23

**Comment by resi23** [comment]
already thinking about updating your server to a current version?

**Comment by masood151117** [comment]
> already thinking about updating your server to a current version?

I will not update the server to 1.21 for six months. Also, most of the plugins were not updated to 1.21.4 in spigotmc yet

**Comment by masood151117** [comment]
I think that customimages should be compatible with 1.20.6 because it is not an old version. If you can give me the fixed version 2.3.3 that you introduced to others in the discussion, or update the current version if you can.

**Comment by Andavin** [comment]
@PurpleEastZQ Thanks for identifying this. I don't want to release a new version on Spigot for this fix alone, but here's the updated file:

[images-2.5.3.jar with bug fixed](https://www.dropbox.com/scl/fi/ardn8o0h7dafwi0apv03a/images-2.5.3.jar?rlkey=sipvgco8exm4a6ogfbnd8sga1&st=8kjiebe2&dl=1)
